lib/utils/sortingUtils.js

```diff
@@ -0,0 +1,103 @@
+
+export function sortByType(array, type) {
+
+  // A custom sort function that sorts elements with $type equal to "type" to the front
+  return array.sort((a, b) => {
+    if (a.$type == type && b.$type != type) {
+      return -1;
+    }
+    if (a.$type != type && b.$type == type) {
+      return 1;
+    }
+    return 0;
+  });
+}
+
+export function sortByPosition(grid, array) {
+  return array.sort((a, b) => {
+    const aPosition = grid.find(a);
+    const bPosition = grid.find(b);
+    if (aPosition[1] !== bPosition[1]) {
+
+      // smallest column to the back to be popped first
+      return Math.sign(bPosition[1] - aPosition[1]);
+    } else {
+
+      // highest row to the back to be popped first
+      return Math.sign(bPosition[0] - aPosition[0]);
+    }
+  });
+}
+
+export function sortBySplittingMergingAndNone(array) {
+  return array.sort((a, b) => {
+    const isASplitting = (a.outgoing || []).length > 1;
+    const isBSplitting = (b.outgoing || []).length > 1;
+    const hasABoundaryEvents = (a.attachers || []).length > 0;
+    const hasBBoundaryEvents = (b.attachers || []).length > 0;
+    const isAMerging = (a.incoming || []).length > 1;
+    const isBMerging = (b.incoming || []).length > 1;
+    if ((isAMerging || isASplitting || hasABoundaryEvents) && !(isBMerging || isBSplitting || hasBBoundaryEvents)) {
+      return -1;
+    }
+    if (!(isAMerging || isASplitting || hasABoundaryEvents) && (isBMerging || isBSplitting || hasBBoundaryEvents)) {
+      return 1;
+    }
+    if (isASplitting && hasBBoundaryEvents) {
+      return 1;
+    }
+    if (isBSplitting && hasABoundaryEvents) {
+      return -1;
+    }
+    return 0;
+  });
+}
+
+export function sortBySplittingAndMerging(array) {
```

Comment:
@till-stadtler It would be benefitial to explain what this function does, and name the argument in a semantic manner.

Reply:
This function is used for sorting outgoing elements behind an already placed element. Splitting elements or elements with boundary events (essentially also splitting) are sorted to the front, while merging elements are sorted to the back. This helps with the layout of the diagram:
1. Elements are placed according to their "index" when looping through them
2. Splitting elements are "simpler", as in their following elements are probably not placed yet
3. Splitting elements come "earlier" in the diagram than merging elements.

For orga: should I add a comment, make the proposed changes and push another commit to the PR?